src/folio_migration_tools/mapping_file_transformation/order_mapper.py

```diff
@@ -388,3 +351,87 @@ def inject_items_schema_by_ref(submodule_path, github_headers, property: dict):
         except Exception as ee:
             logging.error(ee)
             return {}
+
+    def perform_additional_mapping(self, index_or_id, composite_order):
+        # Get organization UUID from FOLIO
+        composite_order["vendor"] = self.get_folio_organization_uuid(
+            index_or_id, composite_order.get("vendor")
+        )
+
+        # Replace legacy bib ID with instance UUID from map
+        bib_id = composite_order["compositePoLines"][0].pop("instanceId", "")
+        if matching_instance := self.get_folio_intance_uuid(index_or_id, bib_id):
+            composite_order["compositePoLines"][0]["instanceId"] = matching_instance
+
+        return composite_order
+
+    def get_matching_record_from_folio(
+        self,
+        index_or_id,
+        cache: dict,
+        path: str,
+        match_property: str,
+        match_value: str,
+        result_type: str,
+    ):
+        if match_value in cache:
+            return cache[match_value]
+        else:
+            query = f'?query=({match_property}=="{match_value}")'
+            if matching_record := next(
+                self.folio_client.folio_get_all(path, result_type, query), None
+            ):
+                cache[match_value] = matching_record
+                return matching_record
+
+    def get_folio_organization_uuid(self, index_or_id, org_code):
+        if self.organizations_id_map:
+            self.migration_report.add(
+                Blurbs.PurchaseOrderVendorLinking,
+                "Organizations linked using organizations_id_map",
+            )
+            if matching_org := self.organizations_id_map.get(org_code):
+                return matching_org[1]
+
+        if matching_org := self.get_matching_record_from_folio(
+            index_or_id,
+            self.folio_organization_cache,
+            "/organizations-storage/organizations",
+            "code",
+            org_code,
+            "organizations",
+        ):
+            self.migration_report.add(
+                Blurbs.PurchaseOrderVendorLinking,
+                "Organizations not in ID map, linked using FOLIO lookup",
+            )
+            return matching_org["id"]
+
+        else:
+            self.migration_report.add(
+                Blurbs.PurchaseOrderVendorLinking,
+                "RECORD FAILED Organization identifier not in ID map/FOLIO",
+            )
+            raise TransformationRecordFailedError(
+                index_or_id,
+                "No matching Organization for organization identifier",
+                org_code,
+            )
+
+    def get_folio_intance_uuid(self, index_or_id, bib_id):
+        if matching_instance := self.instance_id_map.get(bib_id):
+            self.migration_report.add(
+                Blurbs.PurchaseOrderInstanceLinking,
+                "Instances linked using instances_id_map",
+            )
+            return matching_instance[1]
+        else:
```

Comment:
Not saying it's a good idea, but do we want to consider trying to do a lookup based on folio_uuid-generated UUID for the bibId?